crates/sui-core/src/authority.rs

```diff
@@ -1622,11 +1626,24 @@ impl AuthorityState {
         &self,
```

Comment:
nit: rename this function to add_pending_certificates_from_gossip? Not sure where else this is called.

Reply:
it is called in authority_server as well - if we can't execute a cert received via handle_certificate, we enqueue it there for causal completion.

Reply:
Btw, why did we have to enqueue the failed certificate in authority server? Just to avoid lagging behind?